### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by najibullahjafari**
Here:

- I created a class Classroom with the following.
1. @label instance variable, should be initialized in the constructor.
2. Setter and getter for @label (remember about attr_accessor).

- I created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented.
1. Create the has-many side (a classroom has many students).
2. Create the belongs-to side (a student belongs to a classroom).
3. Make sure that when adding a student to a classroom it also sets the classroom for the student.
4. Make sure that when setting the classroom for a student it also adds it to the classroom's students.

- I created a class Book with the following.
1. @title and @author instance variables, should be initialized in the constructor.
2. Setters and getters for instance variables (remember about attr_accessor).

- I created a class Rental with the following.
1. @date instance variable, should be initialized in the constructor.
2. Setter and getter for @date (remember about attr_accessor).

- I created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented.
1. Create the has-many side of Book and Rental (a book has many rentals).
2. Create the belongs-to side of Rental and Book (a rental belongs to a book).
3. Create the has-many side of Person and Rental (a person has many rentals).
4. Create the belongs-to side of Rental and Person (a rental belongs to a person).
5. Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 
 class Student < Person
-  def initialize(id, age: 0, parent_permission: true, classroom: 'Unknown', name: 'Unknown')
+  attr_accessor :classroom, :name
+
+  def initialize(id, age: 0, parent_permission: true, name: 'Unknown')
     super(id, name, age, parent_permission)
-    @classroom = classroom
+    @classroom = nil
   end
```
Comment (OLIPLICHE) [inline student.rb]: - Great JOB. In reading the requirements, it says: "Make sure that when setting the classroom for a student it also adds it to the classrooms' students". This means that you need to add a [method that serves as the `belongs-to`](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)  side (For this case `a student belongs to a classroom`)
- Please click on the blue link above to understand well
```ruby
 def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './book'
+require './person'
+class Rental
+  attr_accessor :book, :person, :date
+
+  def initialize(book, person, date)
+    @date = date
+    @book = book
+    @person = person
```
Comment (OLIPLICHE) [inline rental.rb:9]: From the requirements: 
a-Create the has-many side of Book and Rental (a book has many rentals).
a-Create the belongs-to side of Rental and Book (a rental belongs to a book).

b-Create the has-many side of Person and Rental (a person has many rentals).
b-Create the belongs-to side of Rental and Person (a rental belongs to a person)

Therefore I would advise you to add these codes: 
```ruby
    book.rentals << self  #The rental is adding itself to the @book achieving the 'belongs-to' association
```
```ruby
    person.rentals << self    #The rental is adding itself to the @person the 'belongs-to' association
```

File: person.rb [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
     @name = name
     @parent_permission = parent_permission
     @age = age
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rentals << rental
+    rental.book.add_rental(rental)
   end
```
Comment (OLIPLICHE) [inline person.rb]: Please can you use this method for adding rentals?
```ruby
 def add_rental(book, date)
    Rental.new(date, book, self)  #The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
  end
```
[See examples through this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @najibullahjafari, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Created the remaining classes for school library 👍 
- Created the associations between classes 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @najibullahjafari 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Well-written PR 👍 
* Association properly wired up 👍 
* No linter errors 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**